Makefile builddir definition broken with setuptools 61+

https://github.com/systemd/python-systemd/blob/b109b7587da104e0bfd0111a9d6144d3a2c91038/Makefile#L11

setuptools 61+ has chnaged the directory name (`build/lib.linux-x86_64-cpython-311` instead of `build/lib.linux-x86_64-3.11`). `"cpython-311"` can be read from `sys.implementation.cache_tag`

This makes the package fail to build in Fedora 37+:

```
+ make PYTHON=/usr/bin/python3 check
/bin/sh: line 1: git: command not found
/usr/bin/python3 setup.py build_ext 
running build_ext
/usr/bin/python3 setup.py build
running build
running build_py
package init file 'systemd/test/__init__.py' not found (or not a regular file)
running build_ext
(cd build/lib.linux-x86_64-3.11 && /usr/bin/python3 -m pytest . ../../docs -v)
/bin/sh: line 1: cd: build/lib.linux-x86_64-3.11: No such file or directory
make: *** [Makefile:60: check] Error 1
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,6 +22,7 @@ jobs:
           "3.8",
           "3.9",
           "3.10",
+          "3.11.0-rc.1",
         ]
     name: Python ${{ matrix.python }}
     steps:

--- a/Makefile
+++ b/Makefile
@@ -7,8 +7,10 @@ VERSION := $(shell $(PYTHON) setup.py --version)
 TESTFLAGS = -v
 
 define buildscript
-import sys,sysconfig
-print("build/lib.{}-{}.{}".format(sysconfig.get_platform(), *sys.version_info[:2]))
+import sys, sysconfig, setuptools
+sversion = int(setuptools.__version__.split(".")[0])
+end = sys.implementation.cache_tag if sversion >= 61 else "{}.{}".format(*sys.version_info[:2])
+print("build/lib.{}-{}".format(sysconfig.get_platform(), end))
 endef
 
 builddir := $(shell $(PYTHON) -c '$(buildscript)')

